### DISCUSSION TITLE
Websocket & SSE

### DISCUSSION THREAD

**Discussion by treagod** [martenframework/marten#133] [Q&A] [answered]
Is there a possibility to use Websocketd or Server Side Renderinng with Marten?

I tried the ReponseStream, but I don't understand how I can spawn a Fiber, hold the connection and only emit data when needed. As far as I understand Marten only emits whole response, but not partial ones

**Comment by ellmetha** [comment]
Hey! 👋 

Marten does not support Websockets at the moment. This is something that I envision we could add to the framework in 2024 though.

Regarding [streaming responses](https://martenframework.com/docs/handlers-and-http/introduction#streaming-responses), which Marten supports, those are intended to be used to "stream" a lengthy response back to the client (eg. a large dynamically generated file). They can't be used to hold a connection forever: when the content is fully streamed, the connection is closed.

Hope that helps!

**Reply by treagod** [reply]
Are you also planning to integrate SSE for simpler use cases in addition to websockets? Or generally make it possible to outsource a request to a fibre so that it no longer blocks?

**Reply by ellmetha** [reply, marked as answer]
Yes, this is also something I plan to add to the framework at some point.
